pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1880,6 +1905,7 @@ static void Object_endTypeContext(JSOBJ Py_UNUSED(obj), JSONTypeContext *tc) {
     GET_TC(tc)->rowLabels = NULL;
     NpyArr_freeLabels(GET_TC(tc)->columnLabels, GET_TC(tc)->columnLabelsLen);
     GET_TC(tc)->columnLabels = NULL;
+    PyObject_Free((void *)GET_TC(tc)->cStr);
```

Comment:
I'm not too sure about this approach - generally the CPython API only wants you to call this function when you've allocated with the corresponding function(s)

https://docs.python.org/3/c-api/memory.html#c.PyObject_Free

> Frees the memory block pointed to by p, which must have been returned by a previous call to [PyObject_Malloc()](https://docs.python.org/3/c-api/memory.html#c.PyObject_Malloc), [PyObject_Realloc()](https://docs.python.org/3/c-api/memory.html#c.PyObject_Realloc) or [PyObject_Calloc()](https://docs.python.org/3/c-api/memory.html#c.PyObject_Calloc). Otherwise, or if PyObject_Free(p) has been called before, undefined behavior occurs.